contrib/trace_propagators/src/main/java/io/opentelemetry/contrib/trace/propagation/B3PropagatorInjector.java

```diff
@@ -0,0 +1,76 @@
+/*
+ * Copyright 2020, OpenTelemetry Authors
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package io.opentelemetry.contrib.trace.propagation;
+
+import io.grpc.Context;
+import io.opentelemetry.context.propagation.HttpTextFormat;
+import io.opentelemetry.internal.Utils;
+import io.opentelemetry.trace.Span;
+import io.opentelemetry.trace.SpanContext;
+import io.opentelemetry.trace.SpanId;
+import io.opentelemetry.trace.TraceId;
+import io.opentelemetry.trace.TracingContextUtils;
+import javax.annotation.concurrent.Immutable;
+
+@Immutable
+final class B3PropagatorInjector {
+  private static final int SAMPLED_FLAG_SIZE = 1;
+  private static final int TRACE_ID_HEX_SIZE = 2 * TraceId.getSize();
+  private static final int SPAN_ID_HEX_SIZE = 2 * SpanId.getSize();
+  private static final int COMBINED_HEADER_DELIMITER_SIZE = 1;
+  private static final int SPAN_ID_OFFSET = TRACE_ID_HEX_SIZE + COMBINED_HEADER_DELIMITER_SIZE;
+  private static final int SAMPLED_FLAG_OFFSET =
+      SPAN_ID_OFFSET + SPAN_ID_HEX_SIZE + COMBINED_HEADER_DELIMITER_SIZE;
+  private static final int COMBINED_HEADER_SIZE = SAMPLED_FLAG_OFFSET + SAMPLED_FLAG_SIZE;
+
+  private final boolean singleHeader;
+
+  B3PropagatorInjector(boolean singleHeader) {
+    this.singleHeader = singleHeader;
+  }
+
+  <C> void inject(Context context, C carrier, HttpTextFormat.Setter<C> setter) {
+    Utils.checkNotNull(context, "context");
+    Utils.checkNotNull(setter, "setter");
+
```

Comment:
Objects.requireNotNull() and everywhere.

Reply:
Done, anyway i'll add another PR that removes some non native implementations.